python/lib/communication/dmod/communication/scheduler_request.py

```diff
@@ -50,7 +64,10 @@ def __init__(self, model_request: MaaSRequest, user_id: str, cpus: Optional[int]
         else:
             self.memory_unset = False
             self.memory = mem
-        self.allocation_paradigm: str = allocation_paradigm
+        if isinstance(allocation_paradigm, str) and len(allocation_paradigm.strip()) > 0:
+            self.allocation_paradigm: str = allocation_paradigm
+        else:
+            self.allocation_paradigm: str = self.default_allocation_paradigm_str()
```

Comment:
Yes, _JobAllocationParadigm_ is defined within the _scheduler_ package.  The first part of the issue with using it here is that, if _communication_ imports _scheduler_, I think we start to have circular dependency issues.  

Secondarily, I think  _JobAllocationParadigm_ really does belongs in _scheduler_, as opposed to _communication_.  Moving it could be a technical solution to the first problem, but I think that winds up being bad for the design.  So, the simple work around was just to handle as a string in _communication_.